File: app/src/main/java/com/worldventures/dreamtrips/modules/reptools/service/command/UnlikeSuccessStoryCommand.java
package com.worldventures.dreamtrips.modules.reptools.service.command;

import com.worldventures.dreamtrips.R;
import com.worldventures.dreamtrips.api.success_stories.LikeSuccessStoryHttpAction;
import com.worldventures.dreamtrips.api.success_stories.UnlikeSuccessStoryHttpAction;
import com.worldventures.dreamtrips.api.success_stories.model.ImmutableSuccessStory;
import com.worldventures.dreamtrips.core.api.action.ApiActionCommand;

import io.techery.janet.command.annotations.CommandAction;


@CommandAction
public class UnlikeSuccessStoryCommand extends ApiActionCommand<UnlikeSuccessStoryHttpAction, Void> {

   int id;

   public UnlikeSuccessStoryCommand(int id) {
      this.id = id;
   }

   @Override
   protected Void mapHttpActionResult(UnlikeSuccessStoryHttpAction httpAction) {
      return null;
   }

   @Override
   protected UnlikeSuccessStoryHttpAction getHttpAction() {
      return new UnlikeSuccessStoryHttpAction(id);
   }

   @Override
   protected Class<UnlikeSuccessStoryHttpAction> getHttpActionClass() {
      return UnlikeSuccessStoryHttpAction.class;
   }

   @Override
   public int getFallbackErrorMessage() {
      return R.string.error_fail_to_unlike_success_story;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/reptools/command/UnlikeSuccessStoryCommand.kt
package com.worldventures.dreamtrips.social.service.reptools.command

import com.worldventures.core.service.command.api_action.ApiActionCommand
import com.worldventures.dreamtrips.R
import com.worldventures.dreamtrips.api.success_stories.UnlikeSuccessStoryHttpAction

import io.techery.janet.command.annotations.CommandAction

@CommandAction
class UnlikeSuccessStoryCommand(private val id: Int) : ApiActionCommand<UnlikeSuccessStoryHttpAction, Int>() {

   override fun getHttpAction(): UnlikeSuccessStoryHttpAction = UnlikeSuccessStoryHttpAction(id)

   override fun getHttpActionClass(): Class<UnlikeSuccessStoryHttpAction> = UnlikeSuccessStoryHttpAction::class.java

   override fun getFallbackErrorMessage() = R.string.error_fail_to_unlike_success_story

   override fun mapCommandResult(httpCommandResult: Int?) = id
}
